### PR TITLE
TSK-1544: Throw error for unknown query parameter

### PR DESCRIPTION
**Pull request by gitgoodjhe**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

**Comment by gitgoodjhe** [comment]
https://sonarcloud.io/dashboard?branch=TSK-1544&id=gitgoodjhe_taskana

File: history/taskana-simplehistory-rest-spring/src/test/java/pro/taskana/simplehistory/rest/TaskHistoryEventControllerIntTest.java [hunk 1]
```diff
@@ -31,6 +31,14 @@
 @TaskanaSpringBootTest
 class TaskHistoryEventControllerIntTest {
 
+  private static final ParameterizedTypeReference<TaskHistoryEventPagedRepresentationModel>
+      TASK_HISTORY_EVENT_PAGED_REPRESENTATION_MODEL_TYPE =
+          new ParameterizedTypeReference<TaskHistoryEventPagedRepresentationModel>() {};
+
+  private static final ParameterizedTypeReference<TaskHistoryEventRepresentationModel>
+      TASK_HISTORY_EVENT_REPRESENTATION_MODEL_TYPE =
+          new ParameterizedTypeReference<TaskHistoryEventRepresentationModel>() {};
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-rest-spring/src/test/java/pro/taskana/simplehistory/rest/TaskHistoryEventControllerIntTest.java:40]: Dang! I don't like this Syntax. We have to talk about this in our cop. 
This is just a reminder for me to add the entry.

File: history/taskana-simplehistory-rest-spring/src/test/java/pro/taskana/simplehistory/rest/TaskHistoryEventControllerIntTest.java [hunk 2]
```diff
@@ -256,11 +264,34 @@ void should_GetSpecificTaskHistoryEventWithDetails_When_SingleEventIsQueried() {
...
             restHelper.defaultRequest(),
-            ParameterizedTypeReference.forType(TaskHistoryEventRepresentationModel.class));
+            TASK_HISTORY_EVENT_REPRESENTATION_MODEL_TYPE);
 
     assertThat(response.getBody()).isNotNull();
     assertThat(response.getBody().getDetails()).isNotNull();
   }
 
+  @Test
+  void should_ThrowException_When_ProvidingInvalidFilterParams() {
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-rest-spring/src/test/java/pro/taskana/simplehistory/rest/TaskHistoryEventControllerIntTest.java:274]: This is a duplicate. Please check `should_ReturnBadStatusErrorCode_When_InvalidQueryParameterIsUsed`

File: rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java [hunk 3]
```diff
@@ -940,6 +940,29 @@ void should_ThrowNotAuthorized_When_UserHasNoAuthorizationOnTask() {
         .isEqualTo(HttpStatus.FORBIDDEN);
   }
 
+  @Test
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/task/rest/TaskControllerIntTest.java:943]: This is a duplicate. Please check `testThrowsExceptionIfInvalidFilterIsUsed`

File: rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketAccessItemControllerIntTest.java [hunk 4]
```diff
@@ -135,6 +138,29 @@ void should_DeleteAllAccessItemForUser_ifValidAccessIdOfUserIsSupplied() {
     assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
   }
 
+  @Test
+  void should_ThrowException_When_ProvidingInvalidFilterParams() {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketAccessItemControllerIntTest.java:142]: This is a duplicate. Please check `testThrowsExceptionIfInvalidFilterIsUsed`

File: rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java [hunk 5]
```diff
@@ -287,11 +297,33 @@ void testGetWorkbasketDistributionTargets() {
...
-                DistributionTargetsCollectionRepresentationModel.class));
+            DISTRIBUTION_TARGETS_COLLECTION_REPRESENTATION_MODEL_TYPE);
     assertThat(response.getBody()).isNotNull();
     assertThat(response.getBody().getLink(IanaLinkRelations.SELF)).isNotNull();
     assertThat(response.getHeaders().getContentType()).isEqualTo(MediaTypes.HAL_JSON);
     assertThat(response.getBody().getContent()).hasSize(4);
   }
+
+  @Test
+  void should_ThrowException_When_ProvidingInvalidFilterParams() {
```
Comment (mustaphazorgati) [inline rest/taskana-rest-spring/src/test/java/pro/taskana/workbasket/rest/WorkbasketControllerIntTest.java:308]: This is a duplicate. Please check `testThrowsExceptionIfInvalidFilterIsUsed`

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**